### DISCUSSION TITLE
Secure boot with systemd-boot (and sd-stub?)

### DISCUSSION THREAD

**Discussion by Torxed** [archlinux/archinstall#1546] [Ideas]
I could only find a "Secure Boot" ticket for GRUB, so I'll create this discussion on how to achieve secure boot with systemd as well.
Since the release of [systemd 252 *(2022-10-31)*](https://github.com/systemd/systemd/blob/main/NEWS) a new tool called `systemd-measure` was introduced that could potentially create TPM based EFI stubs utilizing secure boot *(if I've interpreted things correctly.. I'm writing this down to remind myself to check back on this topic as I haven't tested it yet)*.

To quote another section that might be useful:
> `sd-stub` now accepts *(and passes to the initrd and then to the full OS)* new PE sections `.pcrsig` and `.pcrkey` that can be used to embed signatures of expected PCR values, to allow sealing secrets via the TPM2 against pre-calculated PCR measurements.

> `sd-boot` can automatically enroll SecureBoot keys from files found on
> the ESP. This enrollment can be either automatic *('force' mode)* or
> controlled by the user *('manual' mode)*. It is sufficient to place the
> SecureBoot keys in the right place in the ESP and they will be picked
> up by sd-boot and shown in the boot menu.

**Comment by Foxboron** [comment]
My general suggestion would be to not do anything about this until Arch support Secure Boot. This would rely on the `shim` and `MOK` concepts and anything `sd-boot` touches regarding self-enrolled keys is simply not meant for `archinstall` usage.

We would most likely need to provide some scripts for this anyway.

https://lists.archlinux.org/pipermail/arch-releng/2019-January/003892.html

https://gitlab.archlinux.org/archlinux/archiso/-/issues/69

**Reply by Torxed** [reply]
Correct, the approach would have been `shim` + `MDK`. But we've purposefully held off on this until there's official tooling or otherwise easier to enroll. I was hoping FIDO2 devices and `sd-boot` would play well together, but I haven't found a way yet.
